Source/JavaScriptCore/dfg/DFGByteCodeParser.cpp

```diff
@@ -953,6 +953,14 @@ class ByteCodeParser {
             auto* fuzzerAgent = m_vm->fuzzerAgent();
             if (UNLIKELY(fuzzerAgent))
                 return fuzzerAgent->getPrediction(codeBlock, codeOrigin, prediction) & SpecBytecodeTop;
+
+            // If this instruction is derived from op_call_ignore_result, then we do not need to care about the result's prediction.
+            // Let's just return SpecFullTop to avoid SpecNone related ForceOSRExit.
+            auto instruction = codeBlock->instructions().at(codeOrigin.bytecodeIndex().offset());
+            OpcodeID opcodeID = instruction->opcodeID();
+            if (opcodeID == op_call_ignore_result)
+                return SpecFullTop;
+
```

Comment:
This is necessary to be here since it is possible that the following tail-call's caller is `op_call_ignore_result`. So we need to check this inside `getValueProfilePredictionFromForCodeBlockAndBytecodeOffset`.